### PR TITLE
Secure CI docs

### PR DESCRIPTION
**Pull request by aimurphy**
* Secure CI section
* Secure connect feature
* Fix some images and formatting in other topics

### CODE REVIEW

File: docs/continuous-integration/secure-ci/security-hardening.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
```
Comment (aimurphy) [inline docs/continuous-integration/secure-ci/security-hardening.md:1]: @rustd 

Secure CI section with General security hardening advice, secure connect, and other info. Includes links to OIDC, SLSA, etc.

File: docs/continuous-integration/secure-ci/secure-connect.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
```
Comment (aimurphy) [inline docs/continuous-integration/secure-ci/secure-connect.md:1]: @HenAmar This is the new secure connect topic.

File: docs/continuous-integration/secure-ci/secure-connect.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+sidebar_position: 20
+---
+
+:::note
+
+Currently, Secure Connect for Harness Cloud is behind the feature flag `CI_SECURE_TUNNEL`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+
+:::
+
+To use [Harness Cloud build infrastructure](/docs/continuous-integration/use-ci/set-up-build-infrastructure/use-harness-cloud-build-infrastructure) in firewalled environments, such as a corporate network, you can create a secure connect tunnel between the Harness FRP server and an FRP client running in your environment. This allows you to use Harness Cloud build infrastructure with privately-hosted assets, such as internal artifacts repositories and on-premises code repositories. Secure connect is a faster alternative to allowlisting IPs.
```
Comment (HenAmar) [inline docs/continuous-integration/secure-ci/secure-connect.md:13]: "the Harness FRP server and an FRP client" --> the Harness Cloud network to you private network.

File: docs/continuous-integration/secure-ci/secure-connect.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+To configure secure connect:
+
+1. Run the FRP client (FRPC) in your firewalled environment. Where you run the client depends on what assets need to securely connect to Harness and your environment's network configuration.
```
Comment (HenAmar) [inline docs/continuous-integration/secure-ci/secure-connect.md:17]: "FRP client (FRPC)" --> docker client

File: docs/continuous-integration/secure-ci/secure-connect.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+   Compatible connectors include:
+   * [GitHub connectors](/docs/platform/connectors/code-repositories/ref-source-repo-provider/git-hub-connector-settings-reference)
+   * [GitLab connectors](/docs/platform/connectors/code-repositories/ref-source-repo-provider/git-lab-connector-settings-reference)
+   * [Bitbucket connectors](/docs/platform/connectors/code-repositories/ref-source-repo-provider/bitbucket-connector-settings-reference)
+   * [Git connectors](/docs/platform/connectors/code-repositories/ref-source-repo-provider/git-connector-settings-reference)
+   * [Docker connectors](/docs/platform/connectors/cloud-providers/ref-cloud-providers/docker-registry-connector-settings-reference)
+   * [GCP connectors](/docs/platform/connectors/cloud-providers/connect-to-google-cloud-platform-gcp)
+   * [AWS connectors](/docs/platform/connectors/cloud-providers/add-aws-connector)
+
+Once enabled, traffic related to the configured connector is tunneled through the FRPC.
```
Comment (HenAmar) [inline docs/continuous-integration/secure-ci/secure-connect.md:39]: I would also note the the user can use our HARNESS_HTTP_PROXY and HARNESS_HTTPS_PROXY environment variables
. 

If the user wish to tunnel other client via our secure tunnel (e.g curl command).  The secure connect env variables are available for him. 

curl -x HARNESS_HTTPS_PROXY endpoint_url_goe_here.

**Pull request merged**